src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -356,10 +355,6 @@ void ReplicatedMergeTreeQueue::updateStateOnQueueEntryRemoval(
                 LOG_TEST(log, "Removing drop range from current and virtual parts {}", *drop_range_part_name);
                 current_parts.remove(*drop_range_part_name);
             }
-
-            virtual_parts.remove(*drop_range_part_name);
```

Comment:
Looks like this line is still needed

Reply:
There is similar action during inserting into queue:

https://github.com/ClickHouse/ClickHouse/blob/71dc0f0616950a30cf7880d8ff9d95446acf0824/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp#L243

Reply:
Yes, but it's only for DROP PART, we also need to remove virtual part of DROP PARTITION.

Reply:
I see, you are right, thanks for clarification.